src/librustc_parse/parser/mod.rs

```diff
@@ -443,7 +443,9 @@ impl<'a> Parser<'a> {
     crate fn unexpected<T>(&mut self) -> PResult<'a, T> {
         match self.expect_one_of(&[], &[]) {
             Err(e) => Err(e),
-            Ok(_) => unreachable!(),
+            // We can get `Ok(true)` from `recover_closing_delimiter`
+            // which is called in `expected_one_of_not_found`.
+            Ok(_) => FatalError.raise(),
```

Comment:
Remind me -- this is used in the parser as a form of panic? I remember seeing some pretty bad output that resulted from using `FatalError.raise()` on occasion (I think rustc just failed with no output at all) -- can we not use some other method here?

Reply:
Anyway, if this is the "current error handling mechanism" seems fine. I'm just asking for my own edification.

Reply:
It says "stop the world". The reason we aggressively removed their usages was to enable recovery and plow ahead to type checking even in the face of malformed code. It is a valid strategy to stop the world when a condition that could only have been caused by prior malformed code is met, as at that point continuing with our recovery is a liability.